Sources/Commands/SwiftTestTool.swift

```diff
@@ -652,61 +641,48 @@ final class TestRunner {
         return args
     }
 
-    /// Executes the tests without printing anything on standard streams.
-    ///
-    /// - Returns: A tuple with first bool member indicating if test execution returned code 0
-    ///            and second argument containing the output of the execution.
-    private func test(testAt testPath: AbsolutePath) -> (Bool, String) {
-        var output = ""
-        var success = false
-        do {
-            // FIXME: The environment will be constructed for every test when using the
-            // parallel test runner. We should do some kind of caching.
-            let env = try constructTestEnvironment(toolchain: toolchain, options: self.options, buildParameters: self.buildParameters)
-            let process = Process(arguments: try args(forTestAt: testPath), environment: env, outputRedirection: .collect, verbose: false)
-            try process.launch()
-            let result = try process.waitUntilExit()
-            output = try (result.utf8Output() + result.utf8stderrOutput()).spm_chuzzle() ?? ""
-            switch result.exitStatus {
-            case .terminated(code: 0):
-                success = true
-#if !os(Windows)
-            case .signalled(let signal):
-                output += "\n" + exitSignalText(code: signal)
-#endif
-            default: break
-            }
-        } catch {
-            diagnostics.emit(error)
+    private func test(at path: AbsolutePath, writeToOutputStream: Bool) -> (Bool, String) {
+        var stdout: [UInt8] = []
+        var stderr: [UInt8] = []
+
+        func makeOutput() -> String {
+            return String(bytes: stdout + stderr, encoding: .utf8)?.spm_chuzzle() ?? ""
         }
-        return (success, output)
-    }
 
-    /// Executes and returns execution status. Prints test output on standard streams.
-    private func test(testAt testPath: AbsolutePath) -> Bool {
```

Comment:
this is the main change, there use to be two variants of test function, one which collects the output and one that prints to standard streams. this consolidates this into one function that always collects, and prints when needed